src/main/scala/fm/http/client/Response.scala

```diff
@@ -92,10 +92,13 @@ final class FullResponse(response: HttpResponse, val body: Array[Byte]) extends
 }
 
 object AsyncResponse {
+  // https://stackoverflow.com/questions/15991173/is-the-content-length-header-required-for-a-http-1-0-response/15995101
+  // If Content-Length is not sent, then we should read until the the connection is closed
   private def expectBodyContent(response: HttpResponse): Boolean = {
-    val hasContentLength: Boolean = HttpUtil.getContentLength(response, -1) > 0
+    val contentLength: Long = HttpUtil.getContentLength(response, -1L)
+
     val isChunked: Boolean = HttpUtil.isTransferEncodingChunked(response)
-    hasContentLength || isChunked
+    contentLength === -1L || contentLength > 0L || isChunked
```

Comment:
Potential problems with this:

 - The `isChunked` flag basically now does nothing since a chunked transfer encoding will not have a content length set.
 - This does not account for HTTP 1xx, 204 or 304 response which are not allowed to have message bodies (RFC 7230 Section 3.3.3 number 1: https://tools.ietf.org/html/rfc7230#section-3.3.3):

```
   1.  Any response to a HEAD request and any response with a 1xx
       (Informational), 204 (No Content), or 304 (Not Modified) status
       code is always terminated by the first empty line after the
       header fields, regardless of the header fields present in the
       message, and thus cannot contain a message body.
```
 - Should this only be applicable if it is an HTTP 1.0 response OR the "Connection: close" header is set?  It seems like this cannot be an HTTP 1.1 persistent connection if the Content-Length is not set and there is no Chunked transfer encoding.

Reply:
This is kind of a bummer that it's a protected method: https://netty.io/4.1/api/io/netty/handler/codec/http/HttpObjectEncoder.html#isContentAlwaysEmpty-H-